mmdeploy/codebase/mmdet3d/deploy/voxel_detection.py

```diff
@@ -0,0 +1,197 @@
+from typing import Any, Dict, Optional, Sequence, Tuple, Union
+
+import mmcv
+import numpy as np
+import torch
+import torch.nn as nn
+from mmcv.parallel import collate, scatter
+from mmdet3d.core.bbox import get_box_type
+from mmdet3d.datasets.pipelines import Compose
+from torch.utils.data import DataLoader, Dataset
+
+from mmdeploy.codebase.base import BaseTask
+from mmdeploy.codebase.mmdet3d.deploy.mmdetection3d import MMDET3D_TASK
+from mmdeploy.utils import Task, get_root_logger
+from .voxel_detection_model import VoxelDetectionModel
+
+
+class VoxelDetectionWrap(nn.Module):
+
+    def __init__(self, model):
+        super(VoxelDetectionWrap, self).__init__()
+        self.model = model
+
+    def forward(self, voxels, num_points, coors):
+        result = self.model(
+            voxel_input=[voxels, num_points, coors], img_metas=[0])
+        return result[0], result[1], result[2]
+
+
+@MMDET3D_TASK.register_module(Task.VOXEL_DETECTION.value)
+class VoxelDetection(BaseTask):
+
+    def __init__(self, model_cfg: mmcv.Config, deploy_cfg: mmcv.Config,
+                 device: str):
+        super().__init__(model_cfg, deploy_cfg, device)
+
+    def init_backend_model(self,
+                           model_files: Sequence[str] = None,
+                           **kwargs) -> torch.nn.Module:
+        from .voxel_detection_model import build_voxel_detection_model
+        model = build_voxel_detection_model(
+            model_files, self.model_cfg, self.deploy_cfg, device=self.device)
+        return model
+
+    def init_pytorch_model(self,
+                           model_checkpoint: Optional[str] = None,
+                           cfg_options: Optional[Dict] = None,
+                           is_onnx_export=False,
+                           **kwargs) -> torch.nn.Module:
+        from mmdet3d.apis import init_model
+        model = init_model(self.model_cfg, model_checkpoint, self.device)
+        if is_onnx_export:
```

Comment:
why need to do this ? BTW, `is_onnx_export` flag could be get inside this function by calling [torch.onnx.is_in_onnx_export](https://pytorch.org/docs/stable/onnx.html?highlight=is_in_onnx_export#torch.onnx.is_in_onnx_export)